Source/WebCore/html/CanvasBase.cpp

```diff
@@ -356,6 +358,88 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
     return ImageBuffer::create(size(), RenderingPurpose::Canvas, 1, colorSpace, pixelFormat, bufferOptions, context);
 }
 
+bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer> pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
+{
+    if (!scriptExecutionContext() || !scriptExecutionContext()->noiseInjectionHashSalt() || wasLastDrawByBitMap || !scriptExecutionContext()->settingsValues().canvasNoiseInjectionEnabled)
+        return false;
+
+    constexpr auto contextString { "Canvas2DContextString"_s };
+    unsigned salt = computeHash<String, uint64_t>(contextString, *scriptExecutionContext()->noiseInjectionHashSalt());
+    constexpr int bytesPerPixel = 4;
+    auto* bytes = pixelBuffer->bytes();
+    HashMap<uint32_t, uint32_t> pixelColorMap;
+    bool wasPixelBufferModified { false };
+
+    for (size_t i = 0; i < pixelBuffer->sizeInBytes(); i += bytesPerPixel) {
+        auto& redChannel = bytes[i];
+        auto& greenChannel = bytes[i+1];
+        auto& blueChannel = bytes[i+2];
+        auto& alphaChannel = bytes[i+3];
+        bool isBlack { !redChannel && !greenChannel && !blueChannel };
+
+        uint32_t pixel = (redChannel << 24) | (greenChannel << 16) | (blueChannel << 8) | alphaChannel;
+        // FIXME: Consider isEquivalentColor comparision instead of exact match
+        if (!pixel || !alphaChannel || !suppliedColors.isValidValue(pixel) || suppliedColors.contains(pixel))
+            continue;
+
+        if (auto color = pixelColorMap.get(pixel)) {
+            alphaChannel = static_cast<uint8_t>(color);
+            blueChannel = static_cast<uint8_t>(color >> 8);
+            greenChannel = static_cast<uint8_t>(color >> 16);
+            redChannel = static_cast<uint8_t>(color >> 24);
+            continue;
+        }
+
+        auto pixelHash = computeHash(salt, redChannel, greenChannel, blueChannel, alphaChannel);
+
+        auto clampedFivePercent = static_cast<int>((static_cast<float>(pixelHash / std::numeric_limits<uint32_t>::max()))*26 - 13);
```

Comment:
Nit - spaces around the `*`.

```suggestion
        auto clampedFivePercent = static_cast<int>((static_cast<float>(pixelHash / std::numeric_limits<uint32_t>::max())) * 26 - 13);
```